quartz_solar_forecast/eval/forecast.py

```diff
@@ -1,89 +1,103 @@
+        print(f"Model loaded from {model_path}")
+    
+    def forecast_single(self, pv_row, nwp_site_df, nwp_source = "ICON"):
+        """
+        pv_row: DataFrame row containing PV site data
+        nwp_site_df: DataFrame with NWP data for the site
+        nwp_source: Source of NWP data (default: "ICON")
+        """
         site = PVSite(
-            latitude=pv_row["latitude"],
-            longitude=pv_row["longitude"],
-            capacity_kwp=pv_row["capacity"],
+            latitude = pv_row["latitude"],
+            longitude = pv_row["longitude"],
+            capacity_kwp = pv_row["capacity"],
         )
-
-        nwp_site_df = nwp_df[
-            (nwp_df["pv_id"] == pv_row.pv_id) & (nwp_df["timestamp"] == pv_row.timestamp)
-        ]
-
-        pv_id = pv_df["pv_id"][i]
-        ts = pv_df["timestamp"][i]
-
-        # format
+        
+        pv_id = pv_row["pv_id"]
+        ts = pv_row["timestamp"]
+        
+        nwp_df_copy = nwp_site_df.copy()
         for c in ["timestamp", "latitude", "longitude", "pv_id"]:
-            if c in nwp_site_df.columns:
-                nwp_site_df = nwp_site_df.drop(columns=c)
-
-        nwp_site_df.set_index("time", inplace=True, drop=True)
-
+            if c in nwp_df_copy.columns:
+                nwp_df_copy = nwp_df_copy.drop(columns = c)
+        
+        nwp_df_copy.set_index("time", inplace = True, drop = True)
+        
         if isinstance(ts, str):
             ts = datetime.fromisoformat(ts)
-
-        # make pv and nwp data from GFS
-        # TODO move this to model
-        print("Making pv and nwp data")
-        nwp_xr = format_nwp_data(df=nwp_site_df, nwp_source=nwp_source, site=site)
-        pv_xr = make_pv_data(site=site, ts=ts)
-
-        # run model
-        print('Running model')
-        pred_df = forecast_v1(nwp_source, nwp_xr, pv_xr, ts, model=model)
-
-        # only select hourly predictions
+            
+        # Preparing data for model
+        nwp_xr = format_nwp_data(df = nwp_df_copy, nwp_source = nwp_source, site = site)
+        pv_xr = make_pv_data(site = site, ts = ts)
+        
+        # Run model
+        pred_df = forecast_v1(nwp_source, nwp_xr, pv_xr, ts, model = self.model)
+        
+        # Processing
         pred_df = pred_df.resample("1H").mean()
         pred_df["horizon_hour"] = range(0, len(pred_df))
         pred_df["pv_id"] = pv_id
-
-        all_predictions.append(pred_df)
-
-    all_predictions = pd.concat(all_predictions)
-    all_predictions['timestamp'] = all_predictions.index
-
-    return all_predictions
+        pred_df['timestamp'] = pred_df.index
+        return pred_df
+    
+    def forecast_batch(self, pv_df, nwp_df, nwp_source = "ICON"):
+        """        
+        pv_df: DataFrame with PV site data
+        nwp_df: DataFrame with NWP data
+        nwp_source: Source of NWP data (default: "ICON")
+        """
+        all_predictions = []
+        
+        for i in range(len(pv_df)):
```

Comment:
I'm not sure this is right? I'm not sure pv_df and nwp_df are not the same length. 
A quick wait to check this, is do the current tests pass with this refactor?